### PR TITLE
migrate gestures to nullsafety

### PR DESCRIPTION
**Pull request by a14n**
## Description

NNBD migration for gestures

## Related Issues

* dart-lang/sdk#40146

## Tests

I added the following tests:

None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/gestures/monodrag.dart
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/monodrag.dart:91]: This one is surprising. I expected this to be non-nullable.

File: packages/flutter/lib/src/gestures/monodrag.dart
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/monodrag.dart:267]: Why's `+=` no good anymore?

File: packages/flutter/lib/src/gestures/recognizer.dart
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/recognizer.dart:169]: I am surprised that the return type can be null here.

File: packages/flutter/lib/src/gestures/recognizer.dart
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/recognizer.dart:299]: Why the `?` when the next line is a non-null check?

Comment (goderbauer) [reply packages/flutter/lib/src/gestures/recognizer.dart:299]: Oh, do getter and setter need to have the exact same type?

**Review by goderbauer — COMMENTED** [review]
Thanks for doing the migration!

File: packages/flutter/lib/src/gestures/lsq_solver.dart
Comment (a14n) [reply packages/flutter/lib/src/gestures/lsq_solver.dart:102]: done

File: packages/flutter/lib/src/gestures/monodrag.dart
Comment (a14n) [reply packages/flutter/lib/src/gestures/monodrag.dart]: Not sure why I made it nullable :-/ Changed

File: packages/flutter/lib/src/gestures/monodrag.dart
Comment (a14n) [reply packages/flutter/lib/src/gestures/monodrag.dart]: I should have made `_pendingDragOffset` nullable at some point and change it again later. Updated.

File: packages/flutter/lib/src/gestures/recognizer.dart
Comment (a14n) [reply packages/flutter/lib/src/gestures/recognizer.dart:169]: It can be null because the catch block doesn't throw and then the result remains null

File: packages/flutter/lib/src/gestures/recognizer.dart
Comment (a14n) [reply packages/flutter/lib/src/gestures/recognizer.dart:299]: >do getter and setter need to have the exact same type?

Almost. The rule is that setter can accept the return of getter. See https://github.com/dart-lang/language/issues/331

File: packages/flutter/lib/src/gestures/monodrag.dart
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/monodrag.dart:272]: This one can go back to `+=` as well then?

**Review by goderbauer — APPROVED** [review]
LGTM after one last nit :)

File: packages/flutter/lib/src/gestures/monodrag.dart
Comment (a14n) [reply packages/flutter/lib/src/gestures/monodrag.dart]: Fixed. Thanks!

**Comment by a14n** [comment]
Blocked by https://github.com/dart-lang/dartdoc/issues/2273 (and _Google testing failed. Please contact a Google developer to investigate._)

**Comment by goderbauer** [comment]
Checked internally, "Google testing" should be a pass now.

**Comment by goderbauer** [comment]
@a14n You'll probably have to rebase this ontop the latest master to get rid of the cirrus failures.

**Pull request merged**
